File: anagram-concatenation/solutions/TL_90_points.py
n = int(input())
s = list(input())
left = 0
for i in range(n):
	word = input()
	right = 0
	for j in word:
		index = left
		while s[index] != j:
			index += 1
		s[index] = '-'
		right = max(right, index)
	print(left + 1, ' ', right + 1, end = ' ')
	left = right + 1

File: anagram-concatenation/solutions/TL_90_points.cpp
#include <iostream>
#include <string>

using namespace std;

int main () {
	int n;
	cin >> n;
	string s;
	cin >> s;
	int left = 0;
	for (int i = 0; i < n; i++){
	    string z;
	    cin >> z;
	    int right = 0;
	    for (int index = 0; index < z.size(); index++){
	        for (int j = left; j < s.size(); j++){
	            if (z[index] == s[j]){
	                s[j] = '-';
	                right = max(right, j);
	                break;
	            }
    	    }   
	    }
	    cout << left + 1 << " " << right + 1 << " ";
	    left = right + 1;
	}

	return 0;
}
